src/bokeh/tile_providers.py

```diff
@@ -27,19 +27,19 @@
     Raises:
         ValueError, if the specified provider can not be found
 
-    Example:
+    **Example:**
 
-        .. code-block:: python
+    .. code-block:: python
```

Comment:
Actually, is this the problem? AFAIK the entire content under the headings must be indented. Does the heading work if this unindent is removed?

Reply:
I am sorry to disappoint you. The indentation doesn't work, see the posted image above. This is way I was working on the sektion. The headline "example" is not recognized and thats way the code block looks like a citation.

Reply:
I was paying some more attation to the page and the complete block is not correct. Normally "Args" should have a gray background and should be translated to "Arguments". I will spend some time to investigate.

Reply:
I found a solution. The correct way is to us the `autofunction` directive in the rst file and add the docstring to the function.

If I do so, the correct output is this:

![grafik](https://github.com/bokeh/bokeh/assets/68053396/897b85af-f3ed-4479-9283-96d7c10a6e27)

Reply:
@mosc9575 sorry to be such a stickler here but I don't like the last commit because it moves off of the information out of the docstring, meaning it will not be available to anyone using `help` or `??` in interactive settings. I think it is important for the actual library to carry a considerable portion of the documentation with itself.

Reply:
I reverted the last commit and added `.. autofunction:: bokeh.tile_providers.get_provider` in the `tile_providers.py` and moved the docstring from the `get_provider` function to the place a docstring belongs.

The result is, that the `tile_providers` module keeps the `__doc__` 

![grafik](https://github.com/bokeh/bokeh/assets/68053396/6fe908b1-e759-439d-8c47-56d16b489999)

and sphinx correctly recognizes all the "Parameters" and "Example" section.